src/main/java/org/traccar/protocol/Mavlink2ProtocolDecoder.java

```diff
@@ -0,0 +1,128 @@
+package org.traccar.protocol;
+
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.DeviceSession;
+import org.traccar.Protocol;
+import org.traccar.model.Position;
+
+import io.netty.channel.Channel;
+
+public class Mavlink2ProtocolDecoder extends BaseProtocolDecoder {
+
+    public Mavlink2ProtocolDecoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object pkt) throws Exception {
+        byte[] packet = (byte[]) pkt;
```

Comment:
You should not use byte array. Please check other binary protocol. For example `GT02` to see how it should be implemented.